src/bun.js/ModuleLoader.zig

```diff
@@ -2143,24 +2142,23 @@ pub const RuntimeTranspilerStore = struct {
         };
     }
 
-    // This is run at the top of the event loop on the JS thread.
-    pub fn drain(this: *RuntimeTranspilerStore) bun.JSExecutionTerminated!void {
+    pub fn runFromJSThread(this: *RuntimeTranspilerStore) void {
         var batch = this.queue.popBatch();
+        var vm: *VirtualMachine = @fieldParentPtr("transpiler_store", this);
+        const event_loop = vm.eventLoop();
+        const global = vm.global;
+        const jsc_vm = vm.jsc;
         var iter = batch.iterator();
         if (iter.next()) |job| {
             // we run just one job first to see if there are more
-            try job.runFromJSThread();
+            job.runFromJSThread() catch |err| global.reportUncaughtExceptionFromError(err);
         } else {
             return;
         }
-        var vm: *VirtualMachine = @fieldParentPtr("transpiler_store", this);
-        const event_loop = vm.eventLoop();
-        const global = vm.global;
-        const jsc_vm = vm.jsc;
         while (iter.next()) |job| {
             // if there are more, we need to drain the microtasks from the previous run
-            try event_loop.drainMicrotasksWithGlobal(global, jsc_vm);
-            try job.runFromJSThread();
+            event_loop.drainMicrotasksWithGlobal(global, jsc_vm) catch return;
+            job.runFromJSThread() catch |err| global.reportUncaughtExceptionFromError(err);
```

Comment:
if termination exception this should be try right?
```suggestion
            try job.runFromJSThread();
```